src/connectors/infinite-hits/connectInfiniteHits.ts

```diff
@@ -264,28 +264,36 @@ const connectInfiniteHits: InfiniteHitsConnector = (
       getWidgetState(uiState, { searchParameters }) {
         const page = searchParameters.page || 0;
 
-        if (!hasShowPrevious || page === 0 || page + 1 === uiState.page) {
+        if (!hasShowPrevious || !page) {
           return uiState;
         }
 
         return {
           ...uiState,
+          // The page in the UI state is incremented by one
+          // to expose the user value (not `0`).
           page: page + 1,
         };
       },
 
       getWidgetSearchParameters(searchParameters, { uiState }) {
-        if (!hasShowPrevious) {
-          return searchParameters;
+        let state = searchParameters;
+
+        if (escapeHTML) {
+          state = searchParameters.setQueryParameters(TAG_PLACEHOLDER);
         }
 
-        const uiPage = uiState.page;
+        if (hasShowPrevious) {
+          const uiPage = uiState.page;
 
-        if (uiPage) {
-          return searchParameters.setQueryParameter('page', uiPage - 1);
+          if (uiPage) {
+            // The page in the search parameters is decremented by one
+            // to get to the actual parameter value from the UI state.
+            state = searchParameters.setQueryParameter('page', uiPage - 1);
+          }
         }
 
-        return searchParameters.setQueryParameter('page', undefined);
+        return state;
```

Comment:
The widget have to enforce a default value like pagination does otherwise its value is controlled by its parent.

Reply:
So you mean setting an arbitrary page of `0`?

Reply:
Yes.

Reply:
f390b1c737e93de0bf6b3aceea1ca4576a1493cd